crates/forge-runner/src/runtime.rs

```diff
@@ -0,0 +1,135 @@
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.hint_handler.context.vm_run_resources.get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.hint_handler.context.vm_run_resources.run_resources()
+    }
+}
+
+impl<'a> HintProcessorLogic for StarknetRuntime<'a> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        self.hint_handler
+            .execute_hint(vm, exec_scopes, hint_data, constants)
+    }
+
+    fn compile_hint(
+        &self,
+        hint_code: &str,
+        ap_tracking_data: &ApTracking,
+        reference_ids: &HashMap<String, usize>,
+        references: &[HintReference],
+    ) -> Result<Box<dyn Any>, VirtualMachineError> {
+        self.hint_handler
+            .compile_hint(hint_code, ap_tracking_data, reference_ids, references)
+    }
+}
+
+pub struct ExtendedRuntime<Handler, Runtime: HintProcessor> {
+    pub extension_handler: Handler,
+    pub extended_runtime: Runtime,
+}
+
+impl<Handler, Runtime: HintProcessor> ResourceTracker for ExtendedRuntime<Handler, Runtime> {
+    fn consumed(&self) -> bool {
+        self.extended_runtime.consumed()
+    }
+
+    fn consume_step(&mut self) {
+        self.extended_runtime.consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.extended_runtime.get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.extended_runtime.run_resources()
+    }
+}
+
+impl<Handler, Runtime: HintProcessor> HintProcessorLogic for ExtendedRuntime<Handler, Runtime> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        self.extended_runtime
+            .execute_hint(vm, exec_scopes, hint_data, constants)
+    }
+
+    fn compile_hint(
+        &self,
+        hint_code: &str,
+        ap_tracking_data: &ApTracking,
+        reference_ids: &HashMap<String, usize>,
+        references: &[HintReference],
+    ) -> Result<Box<dyn Any>, VirtualMachineError> {
+        self.extended_runtime
+            .compile_hint(hint_code, ap_tracking_data, reference_ids, references)
+    }
+}
+
+#[allow(dead_code)]
+enum HintHandlingResult {
+    Forward,
+    Result(HintExecutionResult),
+}
+
+#[allow(dead_code)]
+enum CheatcodeHadlingResult {
+    Forward,
+    Result(Vec<Felt252>),
+}
+
+trait ExtensionLogic {
+    fn override_system_call(_syscall: SyscallSelector) -> HintHandlingResult {
+        HintHandlingResult::Forward
+    }
+
+    fn handle_cheatcode(_selector: &str, _inputs: Vec<Felt252>) -> CheatcodeHadlingResult {
+        CheatcodeHadlingResult::Forward
+    }
```

Comment:
Hard to say what will happen after this PR but you might be in a similar situation that i was, where separate Extensions would require cross-extension state access. Did you take that into account?